src/kernel/src/fs/path.rs

```diff
@@ -286,11 +287,60 @@ impl From<VPathBuf> for String {
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum LookupOp {
+    Lookup,
+    Create,
+    Delete,
+    Rename,
+}
+
+/// Inspired by [`std::path::Component`].
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum VPathComponent<'a> {
+    Root,
+    Dot,
+    DotDot,
+    Normal { name: &'a str },
```

Comment:
I think `Name(&'a str)` looks better. It is also much easier to work with unnamed field than named field.